errors.go

```diff
@@ -134,6 +135,33 @@ func (e *Error) MarshalJSON() ([]byte, error) {
 	})
 }
 
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (e *Error) UnmarshalJSON(data []byte) error {
```

Comment:
Could you create an `errors_test.go` with a simple test to verify that the `MarshalJSON` and `UnmarshalJSON` functions are proper inverses? Something like:

```go
func TestErrorMarshalUnmarshal(t *testing.T) {
	t.Parallel()

	expected := []byte(`{"id":"1","links":{"about":"A"},"status":"500","code":"C","title":"T","detail":"D","source":{"pointer":"PO","parameter":"PA"},"meta":{"K":"V"}}`)

	var e Error
	err := json.Unmarshal(expected, &e)
	is.MustNoError(t, err)

	actual, err := json.Marshal(&e)
	is.MustNoError(t, err)
	is.EqualJSON(t, string(expected), string(actual))
}
```

Reply:
I just added the test that you suggested, considering that the goal is checking if Marshal and Unmarshal are inverses, I I didn't find anything to add.